tests/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects.Tests/AddSqlProjectTests.cs

```diff
@@ -0,0 +1,66 @@
+using Aspire.Hosting;
+
+namespace CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects.Tests;
+
+public class AddSqlProjectTests
+{
+    [Fact]
+    public void AddSqlProject_WithProjectMetadata()
+    {
+        // Arrange
+        var appBuilder = DistributedApplication.CreateBuilder();
+        appBuilder.AddSqlProject<TestProject>("MySqlProject");
+        
+        // Act
+        using var app = appBuilder.Build();
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+        
+        // Assert
+        var sqlProjectResource = Assert.Single(appModel.Resources.OfType<SqlProjectResource>());
+        Assert.Equal("MySqlProject", sqlProjectResource.Name);
+
+        var dacpacPath = sqlProjectResource.GetDacpacPath();
+        Assert.NotNull(dacpacPath);
+        Assert.True(File.Exists(dacpacPath));
+    }
+
+    [Fact]
+    public void AddSqlProject_WithExplicitPath()
+    {
+        // Arrange
+        var appBuilder = DistributedApplication.CreateBuilder();
+        appBuilder.AddSqlProject("MySqlProject").WithDacpac(TestProject.RelativePath);
+        
+        // Act
+        using var app = appBuilder.Build();
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+        
+        // Assert
+        var sqlProjectResource = Assert.Single(appModel.Resources.OfType<SqlProjectResource>());
+        Assert.Equal("MySqlProject", sqlProjectResource.Name);
+
+        var annotation = Assert.Single(sqlProjectResource.Annotations.OfType<DacpacMetadataAnnotation>());
```

Comment:
Use the `TryGetLastAnnotation` or `TryGetAnnotationsOfType`, here's a test example: https://github.com/CommunityToolkit/Aspire/blob/9bb6f515f7b8b1e2e01bf856fd6e67de38d795c6/tests/CommunityToolkit.Aspire.Hosting.Azure.DataApiBuilder.Tests/ContainerResourceCreationTests.cs#L47-L49